WIP: Add an analyzer for bikes/motos parking in Paris (from OpenData)

Phyks opened the pull request:
Hi,

I've come across the OpenData about parking in Paris (https://opendata.paris.fr/explore/dataset/stationnement-voie-publique-emplacements/table/) which seems to be quite complete (see for instance the nice visualization they have on top of it http://capgeo.sig.paris.fr/Apps/Stationnement2Roues/).

I thought it could be worth integrating it in Osmose and it seemed to be quite easy to do. Here is my attempt at it.

I marked the pull request as WIP because this is my very first PR and I'm not sure at all how to proceed to test it with Osmose (and I did not find docs on this, although I may have missed something). :/ I copy-pasted it from the Bordeaux analysis and adapted it accordingly, but I might have missed some things.

Thanks!

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+
+class Analyser_Merge_Bicycle_Parking_FR_Paris(Analyser_Merge):
+    def __init__(self, config, logger = None):
+        self.missing_official = {"item":"8150", "class": 1, "level": 3, "tag":
```
Since we already have a parking item we can reuse it:
http://osmose.openstreetmap.fr/fr/errors/?item=8150
Bu we need to change class, replace 1 by 21.

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+                    tags = {"amenity": "motorcycle_parking"}),
+                conflationDistance = 50,
+                generate = Generate(
+                    static1 = {"amenity": "motorcycle_parking", "bicycle": "yes"},
```
Since the accuracy look very good we can try to match both.

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+
+###########################################################################
+##                                                                       ##
+## Copyrights Frédéric Rodrigo 2014-2016                                 ##
```
Put your own name, not mine.

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+                    "Régime particulier": u"Vélos",
+                },
+                xFunction = lambda x: self.float_comma(x.split(',')[0]),
+                yFunction = lambda y: self.float_comma(x.split(',')[1])
```
lambda y: .... y.split

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+
+class Analyser_Merge_Motorcycle_Parking_FR_Paris(Analyser_Merge):
+    def __init__(self, config, logger = None):
+        self.missing_official = {"item":"8150", "class": 1, "level": 3, "tag":
```
class 31

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+
+class Analyser_Merge_Bicycle_Motorcycle_Parking_FR_Paris(Analyser_Merge):
+    def __init__(self, config, logger = None):
+        self.missing_official = {"item":"8150", "class": 1, "level": 3, "tag":
```
class 41

frodrigo commented:
You can try your code using Docker:
https://github.com/osm-fr/osmose-backend/tree/master/docker

frodrigo commented:
No easy to answer to your problem, but check the OSMOSE_UNLOCKED_UPDATE.
Anyway, the result of the analysis is in the work/results directory. Event if is not uploaded to the frontend.

frodrigo commented on analysers/analyser_merge_bicycle_parking_FR_paris.py:
```diff
@@ -0,0 +1,130 @@
+                    "Régime particulier": u"Vélos",
+                },
+                xFunction = lambda x: self.float_comma(x.split(',')[0]),
+                yFunction = lambda y: self.float_comma(y.split(',')[1])
```
x is 1, and y is 0

frodrigo commented:
Note: there is now docker branch with a rewrite Docker procedure using docker-compose:
https://github.com/frodrigo/osmose-backend/tree/docker/docker

Phyks commented:
Hi,

I updated with the latest comments. I still could not test the code though :/

> No easy to answer to your problem, but check the OSMOSE_UNLOCKED_UPDATE.

I checked and this should be set to `on`. Plus, it seems the error is actually coming from the backend (where for some reason the password value is still `xxx` instead of `foo`). :/


> Anyway, the result of the analysis is in the work/results directory. Event if is not uploaded to the frontend.

Thanks for this! Sadly, it seems the new analysis is not run :/ I don't have any output in `work/results` directory for this analysis and it does not appear in the log output besides the loading analysis part. Note that I updated `osmose_config.py` as well to ensure the analysis is run :/

Thanks,

frodrigo commented:
@jocelynj this new analyser should be allowed in the frontend.

Phyks commented:
Thanks for the merge!

frodrigo commented:
Please, check the results.
http://osmose.openstreetmap.fr/fr/map/#source=34711&item=8150&class=41&zoom=14&lat=48.85076&lon=2.33789&level=1%2C2%2C3&tags=&fixable=
http://osmose.openstreetmap.fr/fr/errors/?source=34711&item=8150&class=41

Phyks commented:
I've already checked a bit the results and it looks good in my opinion. The only concerns I have so far are:

* The OpenData file is very precise and tends to consider any slight separation on the ground as separating parkings (whereas OSM tends to group them together). Here is an example:

![capture d ecran 2018-11-21 a 17 10 43](https://user-images.githubusercontent.com/3856586/48853796-9551e780-edb0-11e8-8f6b-5e09a6a2bbe7.png)

Not sure if this could be handled easily or is really a concern at all.


* The OpenData file seems to be actually of very good quality and the radius for the merge could probably be lowered. The match for shared "2 wheels" (bike and motos) parking could probably be for both "bicycle_parking" and "motorcycle_parking" as discussed previously. This situation can be seen in

![capture d ecran 2018-11-21 a 17 11 28](https://user-images.githubusercontent.com/3856586/48853920-de09a080-edb0-11e8-84e2-9110fc4eb5ce.png)

where the issue raised is an issue of absence of integration whereas it could be an edit of the current bicycle parking.

Overall, I'd consider these are refinement and it seems totally fine and usable for now in my opinion. It has already helped me spot a few missing parkings around me :)

frodrigo commented:
Ok. Thank you. Please make new PR if you wish to change something.
